test/replication-luatest/gh_6033_box_promote_demote_test.lua

```diff
@@ -0,0 +1,325 @@
+local luatest = require('luatest')
+local helpers = require('test.luatest_helpers')
+local cluster = require('test.luatest_helpers.cluster')
+local g = luatest.group('gh-6033-box-promote-demote')
+
+local make_leader = function(server)
+    server:exec(function()
+        box.ctl.promote()
+    end)
+    server:wait_election_state('leader')
+    server:wait_synchro_queue_owner()
+end
+
+local make_follower = function(server)
+    server:exec(function()
+        box.ctl.demote()
+    end)
+    server:wait_election_state('follower')
+end
+
+g.before_each(function(g)
+    g.cluster = cluster:new({})
+
+    g.box_cfg = {
+        election_mode = 'off';
+        replication_timeout = 0.1,
+        replication = {
+            helpers.instance_uri('server_', 1);
+            helpers.instance_uri('server_', 2);
+            helpers.instance_uri('server_', 3);
```

Comment:
I suggest to make it just `server_1`, `server_2` and `server_3`. It is the same as you wrote, but consistent with how you use `server_1`/`server_2`/`server_3` everywhere else as a member of `g`. The same in all the other places where `instance_uri` is used.